modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -1,178 +1,210 @@
-  end
+  def parse_host_resource(data)
+    tab = Rex::Text::Table.new(
+      "Header" => "Web Search Result",
+      "Indent" => 1,
+      "Columns" => ['IP:Port', 'City', 'Country', "Service"]
+    )
 
-  def parse_host_records(records)
-    records.each do |match|
+    data.each do |match|
       host = match['ip']
       port = match['portinfo']['port']
+      city = match['geoinfo']['city']['names']['en']
+      country = match['geoinfo']['country']['names']['en']
+      service = match['portinfo']['service']
+
+      tab << ["#{host}:#{port}", city, country, service]
 
-      report_service(:host => host, :port => port)
-      print_good("Host: #{host} ,PORT: #{port}")
+      report_host(:host => host,
+                  :info => "Port:#{port}",
+                  :comments => 'Added from ZoomEye Host Resource',
+                  ) if datastore['DATABASE']
     end
+    print_line("#{tab}")
   end
 
-  def parse_web_records(records)
-    records.each do |match|
-      host = match['ip'][0]
-      domains = match['domains']
+  def parse_facets(filed, data)
+    # parse facet field,
+    # host resource support field: 'product', 'device','service', 'os', 'port', 'country', 'city'
+    # web resource suppoty field:  "webapp","component","framework","server", "waf","os","country"
+    field_arr = filed.split(',')
+
+    field_arr.each do |item|
+      tab = Rex::Text::Table.new(
+        "Header" => "Facets Search Result",
+        "Indent" => 1,
+        "Columns" => [item, 'count']
+      )
+      data[item].each do |match|
+        name = match['name']
+        number = match['count']
+        tab << [name, number]
+      end
+      print_line("#{tab}")
+    end
+  end
 
-      report_host(:host => host)
-      print_good("Host: #{host}, Domains: #{domains}")
+  def save_raw_data(dork, data, page)
+    filename = dork.gsub(/[^a-zA-Z ]/,'_')
+    ::File.open("#{filename}_#{page}.json", "wb") do |f|
+      f.write(ActiveSupport::JSON.encode(data))
     end
+
   end
 
   def run
-    # check to ensure api.zoomeye.org is resolvable
     unless zoomeye_resolvable?
       print_error("Unable to resolve api.zoomeye.org")
       return
     end
 
-    @zoomeye_token = login(datastore['USERNAME'], datastore['PASSWORD'])
-    if @zoomeye_token.blank?
-      print_error("Unable to login api.zoomeye.org")
-      return
-    end
-
-    # create ZoomEye request parameters
-    dork = datastore['ZOOMEYE_DORK']
+    dork = datastore['DORK']
     resource = datastore['RESOURCE']
-    page = 1
-    maxpage = datastore['MAXPAGE']
-
-    # scroll max pages from ZoomEye
-    while page <= maxpage
-      print_status("ZoomEye #{resource} Search: #{dork} - page: #{page}")
-      results = dork_search(dork, resource, page) if dork
-      break unless match_records?(results)
+    page = datastore['PAGE']
+    apikey = datastore['APIKEY']
+    facets = datastore['FACETS']
+    current_page = 1
+    resp_data = []
+    while current_page <= page
```

Comment:
Once the page number is no longer needed by `save_raw_data` you can remove the counter logic and use Ruby's `1.upto(datastore['MAXPAGE']) do |current_page|`.